Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -175,38 +177,40 @@ namespace ROS2
             entity->Deactivate();
         }
 
-        // Create the joints
-        auto joints = Utils::GetAllJoints(m_model->root_link_->child_links);
-        for (const auto& [name, jointPtr] : joints)
+        if (!m_useArticulations)
         {
-            AZ_Assert(jointPtr, "joint %s is null", name.c_str());
-            AZ_TracePrintf(
-                "CreatePrefabFromURDF",
-                "Creating joint %s : %s -> %s\n",
-                name.c_str(),
-                jointPtr->parent_link_name.c_str(),
-                jointPtr->child_link_name.c_str());
-
-            auto leadEntity = createdLinks.at(jointPtr->parent_link_name.c_str());
-            auto childEntity = createdLinks.at(jointPtr->child_link_name.c_str());
-            // check if both has RigidBody
-            if (leadEntity.IsSuccess() && childEntity.IsSuccess())
+            auto joints = Utils::GetAllJoints(m_model->root_link_->child_links);
+            for (const auto& [name, jointPtr] : joints)
             {
-                AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
-                auto result = m_jointsMaker.AddJointComponent(jointPtr, childEntity.GetValue(), leadEntity.GetValue());
-                if (result.IsSuccess())
+                AZ_Assert(jointPtr, "joint %s is null", name.c_str());
+                AZ_TracePrintf(
+                    "CreatePrefabFromURDF",
+                    "Creating joint %s : %s -> %s\n",
+                    name.c_str(),
+                    jointPtr->parent_link_name.c_str(),
+                    jointPtr->child_link_name.c_str());
+
+                auto leadEntity = createdLinks.at(jointPtr->parent_link_name.c_str());
+                auto childEntity = createdLinks.at(jointPtr->child_link_name.c_str());
+                // check if both has RigidBody
+                if (leadEntity.IsSuccess() && childEntity.IsSuccess())
                 {
-                    m_status.emplace(name, AZStd::string::format("created as %llu", result.GetValue()));
+                    AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
+                    auto result = m_jointsMaker.AddJointComponent(jointPtr, childEntity.GetValue(), leadEntity.GetValue());
+                    if (result.IsSuccess())
```

Comment:
emplacing is common for these branches - take it out and only build the string based on IsSuccess